calico_rkt/calico_rkt.py

```diff
@@ -14,387 +14,473 @@
+
+        :param endpoint
+        :param profile_name
+        :return: None.
+        """
+        if not self._client.profile_exists(profile_name):
+            _log.info("Creating new profile %s.", profile_name)
+            self._client.create_profile(profile_name)
+
+        _log.info("Setting profile %s on endpoint.", profile_name)
+        self._client.set_profiles_on_endpoint(profile_names=[profile_name],
+                                              endpoint_id=endpoint.endpoint_id)
 
-    # Remove the container from the datastore.
-    datastore_client.remove_workload(hostname=hostname,
-                                     orchestrator_id=orchestrator_id,
-                                     workload_id=container_id)
+    def _remove_profile(self, profile_name):
+        """Remove the profile from datastore if it has no endpoints attached.
+
+        :param profile_name: Name of profile to remove.
+        :return: None.
+        """
+        profile_members = self._client.get_profile_members(profile_name)
+        if self._client.profile_exists(profile_name) and len(profile_members) < 1:
+            try:
+                _log.info("Profile %s has no members, removing from datastore.",
+                          profile_name)
+                self._client.remove_profile(profile_name)
+            except KeyError:
+                _log.error("Cannot remove profile %s: Profile not found.",
+                           profile_name)
+                sys.exit(1)
+
+    def _get_endpoint(self):
+        """Gets endpoint matching the container_id.
+
+        Return None if no endpoint is found.
+        Exits with an error if multiple endpoints are found.
+
+        :param container_id:
+        :return: Calico endpoint object if found, None if not found
+        """
+        try:
+            _log.info("Retrieving endpoint that matches container ID %s",
+                      self.container_id)
+            ep = self._client.get_endpoint(hostname=HOSTNAME,
+                                           orchestrator_id=ORCHESTRATOR_ID,
+                                           workload_id=self.container_id)
+        except KeyError:
+            _log.warning("No endpoint found matching ID %s", self.container_id)
+            return
+        except MultipleEndpointsMatch:
+            _log.error("Multiple endpoints found matching ID %s", self.container_id)
+            sys.exit(1)
 
-    _log.info("Removed Calico interface from %s" % container_id)
+        return ep
 
+    def _find_ipam_plugin(self):
+        plugin_type = self.network_config['ipam']['type']
+        plugin_path = ""
+        for path in self.cni_path.split(":"):
+            if os.path.isfile(os.path.abspath(os.path.join(path, plugin_type))):
+                plugin_path = path
+                break
 
-def _set_profile_on_endpoint(endpoint, profile_name):
-    """
-    Configure the calico profile to the endpoint
+        if not plugin_path:
```

Comment:
I'd prefer if this was done when we call `_find_ipam_plugin` rather than in the method itself.  Easier to follow what is going on.